js/src/common/utils/Drawer.js

```diff
@@ -1,20 +1,41 @@
+import { createFocusTrap } from './focusTrap';
+import { throttle } from './throttleDebounce';
+
 /**
  * The `Drawer` class controls the page's drawer. The drawer is the area the
  * slides out from the left on mobile devices; it contains the header and the
  * footer.
  */
 export default class Drawer {
+  /**
+   * @type {import('./focusTrap').FocusTrap}
+   */
+  focusTrap;
+
   constructor() {
     // Set up an event handler so that whenever the content area is tapped,
     // the drawer will close.
-    $('#content').click((e) => {
+    $('#content').on('click', (e) => {
       if (this.isOpen()) {
         e.preventDefault();
         this.hide();
       }
     });
+
+    this.focusTrap = createFocusTrap('#drawer', { allowOutsideClick: true });
+  }
+
+  resizeHandler(e) {
```

Comment:
When are these resizes actually an issue? I can imagine changes in height easily (keyboard open), but why would we have changes in width?

Reply:
See: https://github.com/flarum/core/pull/3018/commits/438577dae7a79f78cd2e6481d112340af5870917

The focus would end up staying trapped in the header elements if the drawer was opened, then the window was resized to be larger. There'd then be no way to close the drawer (and thus escape the focus trap) without making the window smaller again.

Reply:
I discovered, thanks to a kind person in `#extend` in Discord, this: https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/onchange

No need for resize handlers! We can detect when a media query changes state in JS!

```ts
const phoneMaxWidth = document.documentElement.getPropertyValue('--screen-phone-max')
const drawerAailableMediaQuery = window.matchMedia(`(max-width: ${phoneMaxWidth})`);

drawerAailableMediaQuery.addEventListener('change', (e) => {
  if (e.matches) {
    // Is phone
  } else {
    // Is not phone
  }
})
```

Reply:
Isnt that also effectively a resize handler, but one that only responds to the exact change we care about?